extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -578,9 +579,18 @@ private static HttpServerOptions createSslOptions(HttpBuildTimeConfig buildTimeC
         }
 
         ServerSslConfig sslConfig = httpConfiguration.ssl;
-        //TODO: static fields break config
+
         final Optional<Path> certFile = sslConfig.certificate.file;
         final Optional<Path> keyFile = sslConfig.certificate.keyFile;
+        final List<Path> keys = sslConfig.certificate.keyFiles.orElse(new ArrayList<>());
+        final List<Path> certificates = sslConfig.certificate.files.orElse(new ArrayList<>());
+        if (keyFile.isPresent()) {
+            keys.add(keyFile.get());
```

Comment:
This will mutate the injected config. I think this will work fine now, but if you end up using the config in two different places I think this would cause problems. Can you use a new ArrayList for keys so there is no chance the existing config is mutated?

Reply:
Good point, done.